admin/tool/policy/classes/api.php

```diff
@@ -137,6 +142,28 @@ public static function get_policy_version($policyid, $versionid) {
         return $DB->get_record_sql($sql, $params, MUST_EXIST);
     }
 
+    /**
+     * Calculate the status of a particular policy document version.
+     *
+     * @param int $policyid ID of the policy document.
+     * @param int $versionid ID of the policy document revision.
+     * @return string Status of the policy document version.
+     */
+    public static function get_policy_version_status($policyid, $versionid) {
+        $status = static::VERSION_STATUS_DRAFT;
+        $policy = static::list_policies($policyid)[$policyid];
```

Comment:
It is not optimal to load the list over and over again ... Would it be better to get the document and all the versions as the argument?